interop-testing/src/test/java/io/grpc/stub/StubConfigTest.java

```diff
@@ -112,8 +112,29 @@ public void testConfigureDeadlineNanoTime() {
     // New altered config
     assertNotNull(reconfiguredStub.getCallOptions().getDeadlineNanoTime());
     long maxDelta = MILLISECONDS.toNanos(30);
-    long actualDelta = Math.abs(reconfiguredStub.getCallOptions().getDeadlineNanoTime() - deadline);
-    assertTrue(maxDelta + " < " + actualDelta, maxDelta >= actualDelta);
+
+    /**
+     * This test fails randomly from time to time: https://github.com/grpc/grpc-java/issues/1646
+     *
+     * <p>We suspect STW pauses / system jitter to be at fault. Thus, in order to make this test
+     * more reliable, we take the average of 100 runs.
+     *
+     * <p>For example, if on average (without STW/jitter) one computation takes 20ms, then we can
+     * tolerate 30ms * 100 - 20ms * 100 = 1s of STW/jitter for the average to still be below
+     * the maxDelta of 30ms.
+     */
+    long sum = 0;
+    int samples = 100;
+    long maxObservedDelta = 0;
+    for (int i = 0; i < samples; i++) {
+      long delta =
+          Math.abs(reconfiguredStub.getCallOptions().getDeadlineNanoTime() - deadline);
+      sum += delta;
+      maxObservedDelta = Math.max(maxObservedDelta, delta);
+    }
+    long avgDelta = sum / samples;
+    assertTrue(maxDelta + " < " + avgDelta + ", maxObservedDelta: " + maxObservedDelta,
```

Comment:
I added the `maxOberservedDelta` to figure out if its much larger than the `avgDelta`, which MIGHT indicate a STW pause or some system jitter. If not, this MIGHT indicate that travis ci just sometimes assigns us a slow environment?